src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -82,6 +82,16 @@ export type Fiber = Instance & {
   // if this returns multiple values. Such as a fragment.
   output: any, // This type will be more specific once we overload the tag.
 
+  // Singly linked list fast path to the next fiber with side-effects.
+  nextEffect: ?Fiber,
+
+  // The first and last fiber with side-effect within this subtree. This allows
+  // us to reuse a slice of the linked list when we reuse the work done within
+  // this fiber.
+  firstEffect: ?Fiber,
+  lastEffect: ?Fiber,
```

Comment:
No, no reason to use the root.

What do you mean, "Any node can end up being reused and then you need the start of that slice of the list."?